sdk/storage/azure-storage-blobs/src/block_blob_client.cpp

```diff
@@ -220,7 +220,7 @@ namespace Azure { namespace Storage { namespace Blobs {
 
     auto uploadBlockFunc = [&](int64_t offset, int64_t length, int64_t chunkId, int64_t numChunks) {
       Azure::Core::IO::_internal::RandomAccessFileBodyStream contentStream(
-          fileReader.GetHandle(), offset, length);
+          static_cast<void*>(fileReader.GetHandle()), offset, length);
```

Comment:
This is left for storage to address: https://github.com/Azure/azure-sdk-for-cpp/issues/2182